libraries/AP_Follow/AP_Follow.cpp

```diff
@@ -320,6 +315,68 @@ void AP_Follow::handle_msg(const mavlink_message_t &msg)
             _sysid.set(msg.sysid);
             _automatic_sysid = true;
         }
+        updated = true;
+        break;
+    }
+    case MAVLINK_MSG_ID_FOLLOW_TARGET: {
+        // decode message
+        mavlink_follow_target_t packet;
+        mavlink_msg_follow_target_decode(&msg, &packet);
+
+        // ignore message if lat and lon are (exactly) zero
+        if ((packet.lat == 0 && packet.lon == 0)) {
+            return;
+        }
+        // require at least position
+        if ((packet.est_capabilities & (1<<0)) == 0) {
+            return;
+        }
+
+        Location new_loc = _target_location;
+        new_loc.lat = packet.lat;
+        new_loc.lng = packet.lon;
+        new_loc.set_alt_cm(packet.alt*100, Location::AltFrame::ABSOLUTE);
+
+        // FOLLOW_TARGET is always AMSL, change the provided alt to
+        // above home if we are configured for relative alt
+        if (_alt_type == AP_FOLLOW_ALTITUDE_TYPE_RELATIVE &&
+            !new_loc.change_alt_frame(Location::AltFrame::ABOVE_HOME)) {
+            return;
+        }
+        _target_location = new_loc;
+
+        if (packet.est_capabilities & (1<<1)) {
```

Comment:
I wonder if perhaps we should set _target_velocity_ned to zero if the packet does not contain the velocities.  This would protect against the case where the velocities are sent in some packets but then stop for whatever reason.